src/data/finset/basic.lean

```diff
@@ -1587,6 +1587,10 @@ set.ext $ λ _, mem_image.trans set.mem_image_iff_bex.symm
 lemma nonempty.image (h : s.nonempty) (f : α → β) : (s.image f).nonempty :=
 let ⟨a, ha⟩ := h in ⟨f a, mem_image_of_mem f ha⟩
 
+@[simp]
+lemma nonempty.image_iff (f : α → β) : (s.image f).nonempty ↔ s.nonempty :=
+⟨λ ⟨y, hy⟩, by { rcases mem_image.mp hy with ⟨x, hx, _⟩, exact ⟨x, hx⟩, }, λ h, h.image f⟩
```

Comment:
I think you might be able to spell this
```suggestion
⟨λ ⟨y, hy⟩, let ⟨x, hx, _⟩ := mem_image.mp hy in ⟨x, hx⟩, λ h, h.image f⟩
```
to get extra term mode points, whatever those are worth